### ISSUE TITLE
 is an invalid entity ID for dictionary value @ data['snapshot_entities']

### ISSUE THREAD

**Issue by Bluehawk83** [notoriousbdg/Home-AssistantConfig#38]
When trying to create an automation i keep receiving this error:

![SmartSelect_20221117_020552_Chrome](https://user-images.githubusercontent.com/115945343/202328979-4bf63155-a240-4ae5-a455-61412f114fe0.jpg)

The yaml code for this automatisation is:
`alias: Telenet Test pauze
description: ""
trigger:
  - platform: state
    entity_id:
      - media_player.telenettvbox
      - media_player.tvliving
    from: playing
    to: paused
condition: []
action:
  - service: scene.create
    data:
      scene_id: voorpauze
      snapshot_entities: light.spots_keuken light.lichtbak_keuken
  - delay:
      hours: 0
      minutes: 0
      seconds: 2
      milliseconds: 0
  - service: scene.turn_on
    target:
      entity_id: scene.tv_pauze
    metadata: {}
  - wait_for_trigger:
      - platform: state
        entity_id:
          - media_player.telenettvbox
          - media_player.tvliving
        from: paused
        to: playing
  - service: scene.turn_on
    target:
      entity_id: scene.voorpauze
    metadata: {}
mode: single
`
I am using: 
![SmartSelect_20221117_021829_Chrome](https://user-images.githubusercontent.com/115945343/202330605-c4bd235d-aa68-4841-8832-72f04f3d79a9.jpg)

Logbook errors:
![SmartSelect_20221117_022239_Chrome](https://user-images.githubusercontent.com/115945343/202330897-b12be2e6-40f7-47e5-bfc3-935f717e1306.jpg)
![SmartSelect_20221117_022215_Chrome](https://user-images.githubusercontent.com/115945343/202330902-1ef0a7db-7eca-4641-b403-ef78694e3e42.jpg)

Can anyone help me with this please?

Thanks in advance!

**Comment by raoulteeuwenPE** [comment]
I don't use this myself, but checked https://www.home-assistant.io/integrations/scene/ and see another formatting of the "      snapshot_entities:" key, like so

```
      scene_id: before
      snapshot_entities:
      - climate.ecobee
      - light.ceiling_lights

``` 
Could you try that formatting?

**Comment by Bluehawk83** [comment]
Hi there,

Thx for your answer.  I tried several settings,  the format that i am now using is the formatting that was auto created by the GUI for setting automations, but i will try the formatting you suggested by editting the yaml for the automation.   I ve tried so many things, that i cant remember if i allready tried this format as you suggest.   The page you suggest, with info about the scenes i allready saw. And many hours googling. :)   but thank you very much for your answer, i will try this!    It would be very nice if a find a way to use the "snapshot entity" feature as it should.

I was able to use it with 1 zigbee light,  but now with my general lighting that is knx based, i can not take snapshots so i hope there is a solution/fix for this.

Kind regards!

Outlook voor Android<https://aka.ms/AAb9ysg> downloaden
________________________________

**Comment by Bluehawk83** [comment]
Hi,

Here some feedback.   I have tried with the format you suggested, and i receive no more errors so thats a good point.

 it doesnt give an error but also it is not restoring the state of the entities (lights).

I can see the created scene in the settings/scenes.    But cannot edit this and when the scene is activated, the light settings are not correct.



Outlook voor Android<https://aka.ms/AAb9ysg> downloaden
________________________________

**Comment by raoulteeuwenPE** [comment]
What does the trace now show?

**Comment by Bluehawk83** [comment]
Outlook voor Android<https://aka.ms/AAb9ysg> downloaden
________________________________

**Comment by raoulteeuwenPE** [comment]
I don't see anything here in your last reply?

**Comment by Bluehawk83** [comment]
Hi there,

 I have send some screenshots :)

Thank you for your answer!

Outlook voor Android<https://aka.ms/AAb9ysg> downloaden
________________________________

**Comment by raoulteeuwenPE** [comment]
Whereto :-)? Haven't seen any, not sure where to look...

**Comment by Bluehawk83** [comment]
Did you receive them now?  They are copied again in this mail.

Thx

Outlook voor Android<https://aka.ms/AAb9ysg> downloaden
________________________________

**Comment by raoulteeuwenPE** [comment]
No, not seeing them. Not sure if you do, but please check online at https://github.com/notoriousbdg/Home-AssistantConfig/issues/38#issuecomment-1322505680 .

**Comment by Bluehawk83** [comment]
![processed-271241ea-8774-4aa6-b3c7-23c3890b47ec_GJHzfX7T](https://user-images.githubusercontent.com/115945343/203139462-b8a8ada7-6fe4-4d9e-8cef-27f9ebb8b3ba.jpeg)
![processed-2f199443-e10a-41da-bb35-5dbf334dfa05_IGXpZIM3](https://user-images.githubusercontent.com/115945343/203139465-a5ec86c6-7594-4e2b-9e79-56258a439f54.jpeg)
![processed-45340b20-b3c2-4caa-b50b-c61808c6ea44_JzwxJiLk](https://user-images.githubusercontent.com/115945343/203139468-2ae52d30-92f6-4e37-8f76-9f4878609789.jpeg)
![processed-32d70d3a-ea88-4f95-b679-b6034d907477_l0h3ACHl](https://user-images.githubusercontent.com/115945343/203139471-dd57f706-49fc-4074-b288-35b8adbc2697.jpeg)

I am very sorry for this mistake.  I was replying by email instead of replying by this website. 
 And thats where it did go wrong.   Sorry for the unconvenience! 

I think i know where to look for the problem.  Probably it will be a misconfiguration in my knx installation in stead of with this script/automation.   Will first try to fix some ETS settings for the KNX, when my electrician comes by, and will let you know how that did work out..     thank you very much for your time and effort!!

**Comment by raoulteeuwenPE** [comment]
No problem! Looking forward to learn what the problem was/is :-).

**Comment by Bluehawk83** [comment]
I suspect that it has something to do with the knx, read flag that is missing on some group adressess and maybe for the lights, there is no brightness-address, only a state address.  So i think thats why the scene does know if a light was on or off, but not what his setting was..    only guessing from my side..    will figure this out first and will come back to you!    Problem is i am a noob in HA, so it is a lot trial and error..   but i love the concept and project, and i think it is an amazing community with a lot of knowledge and people  willing to help other people!

Outlook voor Android<https://aka.ms/AAb9ysg> downloaden
________________________________
